File: cpp/split-string.cpp
#include <string>
#include <iostream>
#include <vector>

std::vector<std::string> solution(const std:: string &s);

int main() {
    solution("InteriorCrocodileAlligato");
    solution("InteriorCrocodileAlligator");
    return 0;
}

std::vector<std::string> solution(const std::string &s)
{
    std::vector<std::string> result;
    std::string temp = "";
    for(int i = 0; i < s.length(); i += 2){
        temp = s.substr(i,2);
        temp.length() == 2 ? result.push_back(temp) : result.push_back(temp += '_');    
    }
    return result;
}

File: python/split-string.py
def solution(string):
    result = [string[i:i+2] for i in range(0, len(string), 2)]
    last = len(result)-1
    if len(result[last]) < 2 :
        result[last] += "_"
    return result

print(solution("IntereriorCrocodileAlligator"))
